src/ui/components/common/icon.rs

```diff
@@ -0,0 +1,53 @@
+use leptos::*;
+use std::fmt::format;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum IconType {
+  Eye,
+  EyeSlash,
+  Notifications,
+  Donate,
+  Search,
+  Upvote,
+  Downvote,
+  Crosspost,
+  VerticalDots,
+  Report,
+  Comments,
+  Block,
+  Save,
+}
+
+impl IconType {
+  pub fn as_str(&self) -> &'static str {
+    match self {
+      IconType::Block => "block",
+      IconType::Comments => "comments",
+      IconType::Crosspost => "crosspost",
+      IconType::Donate => "donate",
+      IconType::Downvote => "downvote",
+      IconType::Eye => "eye",
+      IconType::EyeSlash => "eye-slash",
+      IconType::Notifications => "notifications",
+      IconType::Report => "report",
+      IconType::Save => "save",
+      IconType::Search => "search",
+      IconType::Upvote => "upvote",
+      IconType::VerticalDots => "vertical-dots",
+    }
+  }
+}
+
+#[component]
+pub fn Icon(
+  #[prop(into)] icon: MaybeSignal<IconType>,
```

Comment:
This doesn't need to be a signal I don't think. Doesn't need state / can be made at compile time.

Reply:
I could see a scenario where there is one `Icon` component that has its icon type change. `MaybeSignal` accepts both signal and non-signal attributes.